### ISSUE TITLE
implicit_return should not trigger when there are subsequent statements

### ISSUE THREAD

**Issue by jklingen** [realm/SwiftLint#3694]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The rule triggers even though there are subsequent statements in a func, e.g.

```swift
    func asdf() -> String {
        return "qwer"
        return "asdf"
    }
```
will trigger the rule and get autocorrected to
```swift
    func asdf() -> String {
        "qwer"
        return "asdf"
    }
```
which effectively changes runtime behavior of the code.

Of course, there is not point in having subsequent statements after a `return` statement. But it is a commonly used trick during debugging a specific situation, to insert an explicit return at the beginning of a func, e.g. to force a specific value to be returned or bypass an operation that's normally executed by the func, for testing purposes.

The behavior can be quite confusing because it is likely that, by the time of autocorrection happening, the developer's focus is back on the simulator or on the code they are about to debug. They don't realize the changed code and are left to wonder why their code still behaves like `"asdf"` had been returned, even though they explicitly told the func to return `"qwer"`

The actual problem after adding the 2nd return is a different one: the code after the 1st return can never execute. But this is not concern of this rule.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
Pods/SwiftLint/swiftlint --fix path/to/MyFile.swift   
Correcting Swift files at paths path/to/MyFile.swift
Correcting 'MyFile.swift' (1/1)
path/to/MyFile.swift:28:9 Corrected Implicit Return
Done inspecting 1 file for auto-correction!
```

### Environment

* SwiftLint version: **0.43.1**
* Installation method used: **CocoaPods**
* Paste your configuration file:

```yml
opt_in_rules:
  - implicit_return
# [...]

```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? **No**
  If so, paste their relative paths and respective contents.
* Which Xcode version are you using (check `xcodebuild -version`)? **Xcode 12.5**
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
    func asdf() -> String {
        // This triggers a violation:
        return "qwer"
        return "asdf"
    }
```
